docs/trace/customizing-the-sdk/README.md

```diff
@@ -359,10 +359,135 @@ using OpenTelemetry;
+
+To register `MyCustomExporter` and `MyCustomService` we can use the
+`ConfigureServices` and `AddExporter` methods:
+
+```csharp
+using var tracerProvider = Sdk.CreateTracerProviderBuilder()
+    .ConfigureServices(services =>
+    {
+        services.AddLogging();
+        services.AddSingleton<MyCustomService>();
+    })
+    .AddExporter<MyCustomExporter>(ExportProcessorType.Batch)
+    .Build();
+```
+
+When using the `Sdk.CreateTracerProviderBuilder` method the `TracerProvider`
+owns its own `IServiceCollection`. It will only be able to see services
+registered into that collection.
+
+**Note** It is important to correctly manage the lifecycle of the
+`TracerProvider`. See [Building a TracerProvider](#building-a-tracerprovider)
+for details.
+
+### Using the OpenTelemetry.Extensions.Hosting package
+
+**Note** If you are authoring an ASP.NET Core application or using the [.NET
+Generic Host](https://learn.microsoft.com/dotnet/core/extensions/generic-host)
+the
+[OpenTelemetry.Extensions.Hosting](../../../src/OpenTelemetry.Extensions.Hosting/README.md)
+package is the recommended mechanism.
+
+```csharp
+var appBuilder = WebApplication.CreateBuilder(args);
+
+appBuilder.Services.AddSingleton<MyCustomService>();
+
+appBuilder.Services.AddOpenTelemetryTracing(builder => builder
+    .AddExporter<MyCustomExporter>(ExportProcessorType.Batch));
+```
+
+When using the `AddOpenTelemetryTracing` method the `TracerProvider` does not
+own its `IServiceCollection` and instead registers into an existing collection
+(typically the collection used is the one managed by the application host). The
+`TracerProviderBuilder` will be able to access all services registered into that
+collection. For lifecycle management, an [IHostedService
+](https://learn.microsoft.com/dotnet/api/microsoft.extensions.hosting.ihostedservice)
+is used to automatically start the `TracerProvider` when the host starts and the
+host will automatically shutdown and dispose the `TracerProvider` when it is
+shutdown.
+
+**Note** Multiple calls to `AddOpenTelemetryTracing` will configure the same
+`TracerProvider`. Only a single `TraceProvider` may exist in an
+`IServiceCollection` \ `IServiceProvider`.
+
+### Dependency injection `TracerProviderBuilder` extension methods
+
+* `AddExporter<T>`: Adds an export processor for the type `T` (must derive from
+  `BaseExporter<Activity>`) into the `TracerProvider`.
+
+* `AddInstrumentation<T>`: Adds instrumentation of type `T` into the
+  `TracerProvider`.
+
+* `AddProcessor<T>`: Adds a processor of type `T` (must derive from
+  `BaseProcessor<Activity>`) into the `TracerProvider`.
+
+* `ConfigureServices`: Registers a callback function for configuring the
+  `IServiceCollection` used by the `TracerProviderBuilder`. **Note**
+  `ConfigureServices` may only be called before the `IServiceProvider` has been
+  created after which point service can no longer be added.
+
+* `ConfigureBuilder`: Registers a callback function for configuring the
```

Comment:
Maybe move this to the end as it's not expected to be used for regular usage?

Reply:
I moved `SetSampler` up so `ConfigureBuilder` is now the last thing. I agree with this fully but I suspect someone will come along and reorder alphabetically again 😄